### ISSUE TITLE
Drafts for new conversations not displayed in conversation list

### ISSUE THREAD

**Issue by FeuRenard** [signalapp/Signal-Android#7612]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When you produce a draft in a new conversation (without existing messages), the conversation is not displayed in the conversation list and neither is the draft. This is also true for MMS groups.

Legacy issues: #7179 & #7035

According to 

### Steps to reproduce
- Tap pencil button (new conversation)
- Choose contact without existing conversation
- enter text in the input field
- leave conversation with back button

**Actual result:** Conversation is not shown in conversation list and neither is the draft.
**Expected result:** Conversation is shown in conversation list with the draft.

### Screenshots
No meaningful screenshot possible.

### Device info

**Device:** Fairphone 2
**Android version:** 6.0.1
**Signal version:** 4.18.0

### Link to debug log
I intentionally omit a debug log because this issue is easily and reliably reproducible.

**Comment by danieldjewell** [comment]
Agreed, this is a pretty significant issue. I can confirm this is happening still as of v4.52.4

Additionally, there's no warning ... drafts seem to be working fine for existing conversation threads. I suspect that the new conversation code doesn't create a conversation history until *after* a message is sent (upon receipt, it wouldn't be an issue)...

Haven't looked at the code yet but the fix might be a bit challenging - until a message is sent, the recipients (and thus whether or not it's a group or single message) aren't locked in potentially causing problems. (It would be necessary, for example, to return to the "new message" screen accessed by the pencil icon, repopulate the recipient fields, and allow the user to further edit that. I know it can be done, it's just a question of complexity.)

**Comment by landry314** [comment]
Yes, this is still an issue in the latest version. If a message has no conversation history then the message text is still saved but no draft is placed in the conversation list.

If you have disappearing messages for most of your messages then this means many conversations will be empty quite often so this bug is actually a frequent problem.

**Comment by landry314** [comment]
why would this not be fixed? it is a glaring annoyance. most conversations have no history when disappearing messages are on by default as they should be. this means most drafts disappear in the conversation list when they should be at the top. this is bizarre unexpected behavior.

**Comment by landry314** [comment]
too bad you would leave this bug in signal

**Comment by landry314** [comment]
wow, since 2018, this bug has been ignored. incredible.

when you start to write a new message to someone with no history (because it was cleared or you have never talked to them yet) then the draft IS saved BUT it is NOT displayed in the conversation list so it becomes invisible.

sometimes you might start a message and then wait for a better time to send it but it is invisible and it is nice to have the draft reminder so it doesnt get forgotten about.
